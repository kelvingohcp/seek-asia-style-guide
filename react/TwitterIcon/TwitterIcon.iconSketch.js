import React from 'react';
import TwitterIcon from './TwitterIcon';

export const symbols = {
  'TwitterIcon props filled true': <TwitterIcon filled={true} />,
  'TwitterIcon props filled false': <TwitterIcon filled={false} />
};
