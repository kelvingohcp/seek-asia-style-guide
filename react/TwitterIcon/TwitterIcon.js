import svgMarkup from './TwitterIcon.svg';
import svgMarkupGeneric from './TwitterIconGeneric.svg';
import svgMarkupFilled from './TwitterIconFilled.svg';

import PropTypes from 'prop-types';
import React from 'react';

import Icon from '../private/Icon/Icon';

export default function TwitterIcon({ filled, generic, ...props }) {
  let markup = svgMarkup;
  if (filled) {
    markup = svgMarkupFilled;
  } else if (generic) {
    markup = svgMarkupGeneric;
  }
  return <Icon markup={markup} {...props} />;
}

TwitterIcon.displayName = 'TwitterIcon';

TwitterIcon.propTypes = {
  filled: PropTypes.bool,
  generic: PropTypes.bool
};

TwitterIcon.defaultProps = {
  filled: false,
  generic: false
};
