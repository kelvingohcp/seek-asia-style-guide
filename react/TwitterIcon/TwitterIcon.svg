<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="currentColor" d="M12 0C5.4 0 0 5.4 0 12s5.4 12 12 12 12-5.4 12-12S18.6 0 12 0zm0 22.2C6.4 22.2 1.8 17.7 1.8 12 1.8 6.4 6.3 1.8 12 1.8c5.5 0 10.2 4.5 10.2 10.2 0 5.5-4.7 10.2-10.2 10.2zM10.5 17c4 0 7.4-3.3 7-7.5.6-.4.9-.8 1.2-1.3-.5.3-.9.5-1.5.5.6-.3.9-.8 1-1.4-.6.3-1 .6-1.7.6-.9-1.1-2.5-1.1-3.4-.2-.6.6-.8 1.5-.6 2.4-2 0-3.8-1.1-5-2.6-.8 1.2-.5 2.7.8 3.3-.5 0-.8-.2-1.2-.3 0 1.2.8 2.3 2 2.4-.5.2-.8.2-1 0 .3 1.1 1.2 1.8 2.3 1.8-.9.6-3.5 1-3.8 1 1.2.8 2.5 1.3 3.9 1.3"/></svg>