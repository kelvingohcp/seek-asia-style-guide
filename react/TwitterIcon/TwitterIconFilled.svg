<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path fill="currentColor" d="M12 0C5.4 0 0 5.4 0 12s5.4 12 12 12 12-5.4 12-12S18.6 0 12 0zm5.5 9.6c.3 4-3.2 7.5-7 7.5-1.4 0-2.7-.5-3.9-1.1.1 0 2.8-.4 3.8-1-1 0-2-.6-2.3-1.8.5.2.8 0 1 0-1.2-.4-2-1.5-2-2.7.5.3.8.4 1.2.4-1.2-.8-1.5-2.3-.8-3.3 1.2 1.7 3 2.5 5.2 2.7-.3-1-.1-2 .6-2.5 1-.9 2.5-.9 3.5.1.6-.1 1-.3 1.7-.6-.2.6-.6 1-1.1 1.3.6 0 .9-.1 1.5-.4-.3.7-.8.9-1.4 1.4z"/></svg>