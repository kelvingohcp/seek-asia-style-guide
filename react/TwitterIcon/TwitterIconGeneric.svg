<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><path d="M31.067 3.2c-1.333 0.8-2.667 1.333-4.133 1.6-4.533-5.067-12.667-0.267-11.2 6.133-5.467-0.267-10.267-2.933-13.467-7.067-1.733 3.067-0.933 7.067 2 9.067-1.067-0.133-2.133-0.4-2.933-0.933v0.133c0 3.2 2.267 6 5.333 6.667-0.933 0.267-2 0.267-2.933 0.133 0.8 2.667 3.2 4.667 6.133 4.667-2.667 2.133-6.267 3.2-9.733 2.8 6.4 4.267 15.2 4 21.333-0.8 4.933-3.867 7.6-10.133 7.333-16.267 1.333-0.933 2.4-2.133 3.333-3.467-1.2 0.533-2.4 0.933-3.733 1.067 1.2-0.933 2.267-2.267 2.667-3.733z"></path></svg>
