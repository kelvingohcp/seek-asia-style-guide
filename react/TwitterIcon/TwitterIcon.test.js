import React from 'react';
import { shallow } from 'enzyme';
import TwitterIcon from './TwitterIcon';

describe('TwitterIcon:', () => {
  it('should render filled', () => {
    const wrapper = shallow(<TwitterIcon filled />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render generic', () => {
    const wrapper = shallow(<TwitterIcon generic />);
    expect(wrapper).toMatchSnapshot();
  });
});
