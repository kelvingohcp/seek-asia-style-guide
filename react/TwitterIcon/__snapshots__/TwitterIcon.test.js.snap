// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwitterIcon: should render filled 1`] = `
<Icon
  className=""
  markup="mock svg"
  svgClassName=""
/>
`;

exports[`TwitterIcon: should render generic 1`] = `
<Icon
  className=""
  markup="mock svg"
  svgClassName=""
/>
`;
